weblab/datasets/views.py

```diff
@@ -230,3 +230,41 @@ def test_func(self):
     def get_success_url(self, *args, **kwargs):
         ns = self.request.resolver_match.namespace
         return reverse(ns + ':list')
+
+
+class DatasetRenameView(LoginRequiredMixin, UserFormKwargsMixin, UserPassesTestMixin, FormMixin, DetailView):
+    template_name = 'datasets/dataset_rename_form.html'
+    context_object_name = 'dataset'
+    """
+    Delete dataset
+    """
+    model = Dataset
+
+    @property
+    def form_class(self):
+        if self.model is Dataset:
+            return DatasetRenameForm
+
+    def _get_object(self):
+        if not hasattr(self, 'object'):
+            self.object = self.get_object()
+        return self.object
+
+    def test_func(self):
+        return self._get_object().is_deletable_by(self.request.user)
```

Comment:
`is_managed_by` might be better, although it's the same effect! `is_editable_by` doesn't yet work for datasets due to a quirk in Django's permission system, because they're not in the entities app. See also #247.